jetty-start/src/main/java/org/eclipse/jetty/start/CommandLineBuilder.java

```diff
@@ -53,87 +54,155 @@ public static String findJavaBin()
         return "java";
     }
 
-    /**
-     * Perform an optional quoting of the argument, being intelligent with spaces and quotes as needed. If a subString is set in quotes it won't the subString
-     * won't be escaped.
-     *
-     * @param arg the argument to quote
-     * @return the quoted and escaped argument
-     * @deprecated no replacement, quoting is done by {@link #toQuotedString()} now.
-     */
-    @Deprecated
-    public static String quote(String arg)
+    private final StringBuilder commandLine = new StringBuilder();
+    private final List<String> args = new ArrayList<>();
+    private final String separator;
+
+    public CommandLineBuilder()
     {
-        return "'" + arg + "'";
+        this(false);
     }
 
-    private List<String> args;
-
-    public CommandLineBuilder()
+    public CommandLineBuilder(boolean multiline)
     {
-        args = new ArrayList<String>();
+        separator = multiline ? (" \\" + System.lineSeparator() + "  ") : " ";
     }
 
-    public CommandLineBuilder(String bin)
+    /**
+     * Quote a string suitable for use with a command line shell using double quotes.
+     * <p>This method applies doubles quoting as described for the unix {@code sh} commands:
+     * Enclosing characters within double quotes preserves the literal meaning of all characters except
+     * dollarsign ($), backquote (`), and backslash (\).
+     * The backslash inside double quotes is historically weird, and serves
+     * to quote only the following characters: {@code $ ` " \ newline}.
+     * Otherwise, it remains literal.
```

Comment:
```suggestion
     * Quote a string suitable for use with a command line shell using strong quotes {@code '}.
     * <p>This method applies strong quoting as described for the unix {@code sh} commands:
     * Enclosing characters within strong quotes preserves the literal meaning of all characters.
```

We don't want to use double quotes with sh / bash / zsh, we want to use strong quotes `'`, as that will not interpret the contents between the quotes.

Examples:

**`sh` on ubuntu**

```
[joakim@hyperion ~]$ which sh
/usr/bin/sh
[joakim@hyperion ~]$ ls -la /usr/bin/sh
lrwxrwxrwx 1 root root 4 Mar 23  2022 /usr/bin/sh -> dash
[joakim@hyperion ~]$ sh -i -l
$ echo "Hello $USER"
Hello joakim
$ echo 'Hello $USER'
Hello $USER
```

**`bash` on ubuntu**

```
[joakim@hyperion ~]$ echo "Hello $USER"
Hello joakim
[joakim@hyperion ~]$ echo 'Hello $USER'
Hello $USER
```

**`zsh` on ubuntu**

```
[joakim@hyperion ~]$ zsh -i -l
joakim@hyperion ~ % echo "Hello $USER"
Hello joakim
joakim@hyperion ~ % echo 'Hello $USER'
Hello $USER
```

Reply:
> @joakime I think what you have been trying to say is: "xargs doesn't grok double quotes like y'd think". Once I worked that out, then OK let's go with single quotes. updated.

It's not xargs doing that. 
It is `bash` doing that, `bash` (and `zsh`) will evaluate on any redirection/pipe before redirection/pipe (unlike `dash`).
This was a design decision by `bash` to get around deficiencies in `sh` not allowing variables to be set on redirection/pipe.